### ISSUE TITLE
s->session_ctx->new_session_cb is not called in TLS 1.3 during for the resumed connections

### ISSUE THREAD

**Issue by yctung** [openssl/openssl#4045]
Hi, I notice the s->session_ctx->new_session_cb is not called (because s->hit == 1) in TLS 1.3 when a client resumes a connection and later get a new session (which is common behavior in TLS1.3). 

Here is the relevant code in ssl_lib.c:

```
void ssl_update_cache(SSL *s, int mode)
{
    int i;

    /*
     * If the session_id_length is 0, we are not supposed to cache it, and it
     * would be rather hard to do anyway :-)
     */
    if (s->session->session_id_length == 0)
        return;

    i = s->session_ctx->session_cache_mode;
    if ((i & mode) //&& (!s->hit) <- here is my change
        && ((i & SSL_SESS_CACHE_NO_INTERNAL_STORE)
            || SSL_CTX_add_session(s->session_ctx, s->session))
        && (s->session_ctx->new_session_cb != NULL)) {
        SSL_SESSION_up_ref(s->session);
        if (!s->session_ctx->new_session_cb(s, s->session))
            SSL_SESSION_free(s->session);
    }
   ....
}
```

I think it is necessary to call the new_session_cb every time when we have a new session, right? Ignore this "(!s->hit)" condition solves my issue (at least in my system), but not sure if it is going to have any side effect.

**Comment by mattcaswell** [comment]
Thanks for your report. I have confirmed this bug. Fix in #4068.

**Comment by yctung** [comment]
Hi Matt,

Thank you for the fix

Yu-Chih
